### PR TITLE
Improved: Migrated from moment timezone to luxon timezone(#2dewerv)

### PR DESCRIPTION
**Pull request by shashwatbangar**
# Related Issues

### Closes #
https://github.com/hotwax/reroute-fulfilment/commit/313ab146d57743b7304283edd4b7ec36b37dad86

### Contribution and Currently Important Rules Acceptance

 [✅] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "axios-cache-adapter": "^2.7.3",
     "core-js": "^3.6.5",
     "http-status-codes": "^2.1.4",
+    "luxon": "^2.4.0",
     "mitt": "^2.1.0",
     "moment": "^2.29.1",
```
Comment (disha1202) [inline package.json:26]: Remove moment.

File: package.json [hunk 2]
```diff
@@ -53,6 +55,7 @@
     "ts-jest": "^26.0.4",
     "typescript": "~4.1.5",
     "vue-cli-plugin-i18n": "^1.0.1",
-    "vue-jest": "^5.0.0-0"
+    "vue-jest": "^5.0.0-0",
+    "vue-loader-v16": "^16.0.0-beta.5.4"
```
Comment (disha1202) [inline package.json:59]: Revert unwanted changes.

File: src/main.ts [hunk 3]
```diff
@@ -42,14 +41,17 @@ app.config.globalProperties.$filters = {
...
     // TODO Make default format configurable and from environment variables
-    return moment(value, inFormat).format(outFormat ? outFormat : 'MM-DD-YYYY');
+    if(inFormat){
+      return DateTime.fromFormat(value, inFormat).toFormat(outFormat ? outFormat : 'MM-DD-YYYY');
+    }
+    return DateTime.fromISO(value).toFormat(outFormat ? outFormat : 'MM-DD-YYYY');
   },
-  formatUtcDate(value: any, inFormat?: string, outFormat?: string) {
+  formatUtcDate(value: any, inFormat?: any, outFormat?: string) {
     // TODO Use Loxon instead
```
Comment (disha1202) [inline src/main.ts:50]: We can remove this TODO now.

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (shashwatbangar) [reply package.json:59]: Done.

File: src/components/ProductListItem.vue [hunk 4]
```diff
@@ -34,7 +34,7 @@ export default defineComponent({
       //TODO need to implement updateCurrentProduct action
       await this.store.dispatch('product/updateCurrentProduct', {product: this.product});
       this.router.push({ path: `/product/${this.product.sku}` })
-    }
+    },
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/components/ProductListItem.vue:37]: Fixed it.

File: src/main.ts (hunk 3 above)
Comment (shashwatbangar) [reply src/main.ts]: Done.

File: src/main.ts [hunk 5]
```diff
@@ -40,16 +39,17 @@ const app = createApp(App)
 // Filters are removed in Vue 3 and global filter introduced https://v3.vuejs.org/guide/migration/filters.html#global-filters
 app.config.globalProperties.$filters = {
   formatDate(value: any, inFormat?: string, outFormat?: string) {
-    // TODO Use Loxon instead
     // TODO Make default format configurable and from environment variables
-    return moment(value, inFormat).format(outFormat ? outFormat : 'MM-DD-YYYY');
+    if(inFormat){
+      return DateTime.fromFormat(value, inFormat).toFormat(outFormat ? outFormat : 'MM-DD-YYYY');
+    }
+    return DateTime.fromISO(value).toFormat(outFormat ? outFormat : 'MM-DD-YYYY');
```
Comment (disha1202) [inline src/main.ts:46]:
```suggestion
    return DateTime.fromISO(value).toFormat(outFormat ? outFormat : 'MM-dd-yyyy');
```

File: src/main.ts [hunk 6]
```diff
@@ -40,16 +39,17 @@ const app = createApp(App)
...
   },
-  formatUtcDate(value: any, inFormat?: string, outFormat?: string) {
-    // TODO Use Loxon instead
+  formatUtcDate(value: any, inFormat?: any, outFormat?: string) {
     // TODO Make default format configurable and from environment variables
     const userProfile = store.getters['user/getUserProfile'];
     // TODO Fix this setDefault should set the default timezone instead of getting it everytiem and setting the tz
-    return moment.utc(value, inFormat).tz(userProfile.userTimeZone).format(outFormat ? outFormat : 'MM-DD-YYYY');
+    return DateTime.utc(value, inFormat).setZone(userProfile.userTimeZone).toFormat(outFormat ? outFormat : 'MM-DD-YYYY')
```
Comment (disha1202) [inline src/main.ts:52]:
```suggestion
    return DateTime.utc(value, inFormat).setZone(userProfile.userTimeZone).toFormat(outFormat ? outFormat : 'MM-dd-yyyy')
```

File: src/main.ts [hunk 7]
```diff
@@ -40,16 +39,17 @@ const app = createApp(App)
...
+    if(inFormat){
+      return DateTime.fromFormat(value, inFormat).toFormat(outFormat ? outFormat : 'MM-DD-YYYY');
+    }
+    return DateTime.fromISO(value).toFormat(outFormat ? outFormat : 'MM-DD-YYYY');
   },
-  formatUtcDate(value: any, inFormat?: string, outFormat?: string) {
-    // TODO Use Loxon instead
+  formatUtcDate(value: any, inFormat?: any, outFormat?: string) {
     // TODO Make default format configurable and from environment variables
     const userProfile = store.getters['user/getUserProfile'];
```
Comment (disha1202) [inline src/main.ts:50]: Access the userTimeZone from state itself instead of using getters

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 5 above)
Comment (shashwatbangar) [reply src/main.ts:46]: Done.

File: src/main.ts (hunk 6 above)
Comment (shashwatbangar) [reply src/main.ts:52]: Done.

**Pull request closed without merging**